apps/src/netsim/NetSimRouterLogModal.html.ejs

```diff
@@ -30,6 +45,69 @@ var showPacketInfo = headerFields.indexOf(Packet.HeaderType.PACKET_INDEX) > -1 &
  * @type {boolean}
  */
 
+function routerLogModeToggleButton() {
+  if (!locals.canToggleRouterLogMode) {
+    return '';
+  }
+
+  var markup = '<button type="button" id="routerlog-toggle" class="pull-right btn btn-primary btn-mini">';
+  if (locals.isAllRouterLogMode) {
+    markup += i18n.logBrowserHeader_toggleMine()
+  } else {
+    markup += i18n.logBrowserHeader_toggleAll()
+  }
+  markup += '</button>';
+  return markup;
+}
+
+/**
+ * Intentionally mirrors NetSimRouterLogModal.cycleTrafficFilterMode_
+ */
+function trafficFilterCycleButton() {
+  if (!locals.canCycleTrafficFilterMode) {
+    return '';
+  }
+  var markup = '<button type="button" id="trafficfilter-cycle" class="pull-right btn btn-primary btn-mini">';
+  if (/^none/.test(locals.currentTrafficFilter)) {
+    markup += i18n.logBrowserHeader_showMyTraffic();
+  } else if (/^with/.test(locals.currentTrafficFilter)) {
+    markup += i18n.logBrowserHeader_showTrafficFromMe();
+  } else if (/^from/.test(locals.currentTrafficFilter)) {
+    markup += i18n.logBrowserHeader_showTrafficToMe();
+  } else {
+    markup += i18n.logBrowserHeader_showAllTraffic();
+  }
+  markup += '</button>';
+  return markup;
+}
+
+function dialogTitle() {
+  var header;
+  if (locals.isAllRouterLogMode) {
+    header = i18n.logBrowserHeader_all();
+  } else {
+    header = i18n.logBrowserHeader_mine();
+  }
+
+  var match = /^(from|to|with) ([\d\.]+)/.exec(locals.currentTrafficFilter);
+  if (match) {
+    if ('from' === match[1]) {
+      header += i18n.logBrowserHeader_trafficFromAddress({
+        address: match[2]
+      });
+    } else if ('to' === match[1]) {
+      header += i18n.logBrowserHeader_trafficToAddress({
+        address: match[2]
+      });
+    } else if ('with' === match[1]) {
```

Comment:
I'm not sure if I'm a fan of the multiple different ways to determine what traffic filter state we're in